### PR TITLE
add marked output to `cabal list-bin`

### LINKED ISSUE #8400: `cabal list-bin` does not produce marked output
Lifted from https://github.com/haskell/cabal/pull/8392#issuecomment-1219171877 
```
$ cabal list-bin cabal '-vnormal +markoutput +nowrap'
Warning: this is a debug build of cabal-install with assertions enabled.
-----BEGIN CABAL OUTPUT-----
Up to date
-----END CABAL OUTPUT-----
/Users/abel/bin/src/cabal/dist-newstyle/build/x86_64-osx/ghc-9.2.4/cabal-install-3.9.0.0/x/cabal/build/cabal/cabal
```
This means the interesting output will not be part of a simple golden value.

The reason is that a plain `putStr` is used for the output: https://github.com/haskell/cabal/blob/a97e1f13bdb28a3f761d55845f14a9072bee17ca/cabal-install/src/Distribution/Client/CmdListBin.hs#L136
That's not in line with convention.  Output should be produced by the predefined API: https://github.com/haskell/cabal/blob/a97e1f13bdb28a3f761d55845f14a9072bee17ca/Cabal/src/Distribution/Simple/Utils.hs#L34-L37
See also:
- https://github.com/haskell/cabal/pull/8396

### PR DESCRIPTION
**Pull request by Cmdv**
Fixes #8400

unsure if using the correct service functions to do this 😄 

---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [x] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: cabal-install/src/Distribution/Client/CmdListBin.hs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE RecordWildCards   #-}
 {-# LANGUAGE TupleSections     #-}
+{-# OPTIONS_GHC -Wno-deferred-out-of-scope-variables #-}
```
Comment (ulysses4ever) [inline cabal-install/src/Distribution/Client/CmdListBin.hs:6]: Hmm, this is a bit unexpected. Is this necessary? Why?

Comment (Cmdv) [reply cabal-install/src/Distribution/Client/CmdListBin.hs:6]: oops sorry didn't spot that!! it was HLS 🤦

**Review by ulysses4ever — APPROVED** [review]
Thanks a lot for the contribution!

**Comment by ulysses4ever** [comment]
Something funny hit the macos-9.4 job: https://github.com/haskell/cabal/actions/runs/3585117132/jobs/6056940056 Will try to restart, but looks like a CI issue. Definitely, nothing to do with the patch.

**Comment by ulysses4ever** [comment]
Oh, this is tracked at #6489

**Comment by Cmdv** [comment]
ah weird I wasn't able to reproduce it locally either and I'm on macos 9.4.2 🤔

**Comment by ulysses4ever** [comment]
Okay, just restarting seemed to have helped. As the author of the issue this is addressing, I invite @andreasabel to review...

**Review by andreasabel — APPROVED** [review]
Thanks, I guess this is what I wanted!

**Comment by andreasabel** [comment]
I am a bit puzzled that no test had to be updated.  Apparently, we are lacking tests for `list-bin`.

**Comment by jneira** [comment]
> I am a bit puzzled that no test had to be updated. Apparently, we are lacking tests for `list-bin`.

And a test should had been added this time imo

**Pull request merged**